File: LeetCode_1/leetCode_1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        //std::unordered_map is better than std::map
        std::unordered_map<int,int> mapping;
        int complement;

        for(int i = 0; i< nums.size(); i++)
        {
            complement = target-nums[i];

            //find if the complement is inside our visited map
            if(mapping.count(complement))   //if true, found and return
            {
                return {mapping[complement],i};
            }

            //if cannot find, add the current number in vector<int> nums to the mapping
            mapping.insert(std::pair<int,int>(nums[i],i));
        }

        //if nothing, return nothing
        return {};
    }
};

//7~20 ms, 10.9 MB

File: LeetCode_1/leetCode_1.py
class Solution:
    def twoSum(self, nums: [int], target: int) -> [int]:
        temp={}
        for i,num in enumerate(nums):
            if( target - num in temp):
                return temp[target-num],  i
            temp[num]=i

'''
if __name__ == '__main__':
    sol = Solution()
    testCase = [3,3,4,1]
    answer = sol.twoSum(testCase, 5)
    print(answer)
'''

'''
48~91ms, 15.4 MB
'''